Tophat/Views/DeviceList.swift

```diff
@@ -11,43 +11,50 @@ import TophatFoundation
 
 struct DeviceList: View {
 	@Environment(\.scenePhase) private var scenePhase
-	@EnvironmentObject private var deviceManager: DeviceManager
+	@Environment(DeviceManager.self) private var deviceManager
+
 	@CodableAppStorage("PinnedDevices") private var pinnedDeviceIdentifiers: [String] = []
-	@State private var otherDevicesExpanded = false
+
+	@State private var cachedDevices: [Device] = []
+	@State private var cachedSimulators: [Device] = []
+
+	@State private var isOtherSimulatorsExpanded = false
 
 	var body: some View {
 		VStack(alignment: .leading, spacing: Theme.Size.menuMargin) {
-			if !primaryDevices.isEmpty {
-				HStack(alignment: .center) {
-					Text("Devices")
-						.sectionHeadingTextStyle()
+			if !devices.isEmpty {
+				DevicePickerSection(
+					title: "Devices",
+					devices: devices,
+					isLoading: deviceManager.isLoading
+				)
 
-					if deviceManager.isLoading {
-						ProgressView()
-							.progressViewStyle(.circular)
-							.controlSize(.mini)
-							.labelsHidden()
-					}
-
-					Spacer()
-				}
-				.padding(.horizontal, Theme.Size.menuPaddingHorizontal)
-				.padding(.top, Theme.Size.menuPaddingVertical)
+				Divider()
+					.padding(.horizontal, Theme.Size.menuPaddingHorizontal)
+			}
 
-				DevicePicker(devices: primaryDevices)
+			if !pinnedSimualators.isEmpty {
+				DevicePickerSection(
+					title: "Simulators",
+					devices: pinnedSimualators,
+					isLoading: deviceManager.isLoading && devices.isEmpty
+				)
 
 				Divider()
 					.padding(.horizontal, Theme.Size.menuPaddingHorizontal)
 			}
 
-			CollapsibleSection("Other Devices", expanded: $otherDevicesExpanded) {
+			CollapsibleSection(
+				pinnedSimualators.isEmpty ? "Simulators" : "Other Simulators",
+				expanded: $isOtherSimulatorsExpanded
+			) {
 				ScrollView(.vertical, showsIndicators: false) {
-					DevicePicker(devices: secondaryDevices)
+					DevicePicker(devices: otherSimulators)
 				}
 				.frame(maxHeight: 240)
 			}
 		}
-		.onChange(of: scenePhase) { newValue in
+		.onChange(of: scenePhase) { oldValue, newValue in
```

Comment:
Is this here just to fix one of the warnings we had?

Reply:
Yes indeed!